src/Features/Core/Portable/PopulateSwitch/PopulateSwitchHelpers.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Semantics;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.PopulateSwitch
+{
+    internal static class PopulateSwitchHelpers
+    {
+        public const string MissingCases = nameof(MissingCases);
+        public const string MissingDefaultCase = nameof(MissingDefaultCase);
+
+        public static bool HasDefaultCase(ISwitchStatement switchStatement)
+        {
+            foreach (var switchCase in switchStatement.Cases)
+            {
+                if (HasDefaultCase(switchCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool HasDefaultCase(ISwitchCase switchCase)
+        {
+            foreach (var clause in switchCase.Clauses)
```

Comment:
This can be optimized by iterating backwards.  VB.NET requires the default case to be the last one, and C# will have it be the last one over 99% of the time.